USAGE.adoc

```diff
@@ -325,6 +325,24 @@ s.caseOf("baz", bc -> {
 s.defaultCase(bc -> bc.assign(ret, bc.loadNull()));
 ----
 
+=== Type conversions (casting)
+
+There are 3 methods for type conversions:
+
+* `checkCast()`
+* `convertPrimitive()`
+* `smartCast()`
+
+The `checkCast()` method emits the `checkcast` instruction, which performs a dynamic type check and a reference conversion.
+If the value that is supposed to be converted is statically known to be of a primitive type, the `checkCast()` method emits a boxing conversion first, before the type conversion.
+
+The `convertPrimitive()` method emits primitive conversion instructions (`i2l`, `d2i` etc.).
+If the value that is supposed to be converted is statically known to be of a primitive wrapper class, the `convertPrimitive()` method emits an unboxing conversion first, before the type conversion.
+
+The `smartCast()` method emits a sequence of instructions to cast the value to the given target type, using boxing, unboxing, primitive and reference conversions.
+For example, it allows casting `java.lang.Integer` to `java.lang.Long` by an unboxing conversion, followed by a primitive conversion, followed by a boxing conversion.
+This method is _not_ equivalent to the casting conversion described by Java Language Specification; it is a superset.
```

Comment:
Is the new method a superset of the other two? If so could the other two be deprecated? If not could that fact be documented along with the specific differences?

Reply:
It is a superset indeed.

I don't really see a reason for deprecating the other two methods. Or, I mean, I don't see anything obsolete about them. Their javadoc points to `smartCast()` as a "smarter" alternative, which I think should be good enough.

Reply:
-1 for deprecation. Sometimes it's not necessary to be that "smart" and the simple variants are good enough. And the javadoc is IMO clear.